pkg/catalog/catalog.go

```diff
@@ -45,3 +49,252 @@ func newAttributeDef(name string, typ types.Type, isPrimary bool) engine.TableDe
 		},
 	}
 }
+
+// consume a set of entries and return a command and the remaining entries
+func ParseEntryList(es []*api.Entry) (any, []*api.Entry, error) {
+	if len(es) == 0 {
+		return nil, nil, nil
+	}
+	e := es[0]
+	if e.DatabaseId != MO_CATALOG_ID {
+		return e, es[1:], nil
+	}
+	switch e.TableId {
```

Comment:
I remember there are 3 tables?  Only two left now?

Reply:
3,  mo_database, mo_tables, mo_columns.

Reply:
Did you just handle 2?    Let's handle 3 explicitly and default case, panic with an error.

Reply:
I handled three of them ， create database generates tuples written to mo_database, and create database generates tuples written to mo_tables and mo_columns.
 consumes logtail does not care whether the specific table is a catalog or not,，https://github.com/matrixorigin/matrixone/blob/main/pkg/vm/engine/disttae/logtail.go#L81。

Reply:
Then when I read the catalog information again, I pass in its specific id. for example: https://github.com/matrixorigin/matrixone/blob/5583ac15a3f690b3ce15815702174d003b394221/pkg/vm/engine/disttae/txn.go#L28

Reply:
If mo_columns should never reach here, panic an internal error as well.